src/extras/named-register.js

```diff
@@ -28,10 +28,12 @@
     if (typeof name !== 'string')
       return register.apply(this, arguments);
     const define = [deps, declare];
-    this.registerRegistry[name] = define;
+    tmpDefine = define;
+    this.registerRegistry[name] = System.getRegister();
+    tmpDefine = null;
```

Comment:
Why is this indirection needed? If `getRegister()` will just return `tmpDefine` directly what it is it achieving by running the hook? Is it to communicate with other hooks, and which ones?

Reply:
Yes it's to give other extras a chance to modify the define. None of the supported extras would be impacted by this (name-exports ignores System.register modules), but user-defined extras would be impacted.

If we'd rather not worry about the user-defined extras with this edge case, I can remove it. I added it for consistent behavior with getRegister

Reply:
Yes I would rather just set it up directly than make it hookable, at least until / if there is a specific use case for it.

Reply:
👍 updated